pkg/controller/jobframework/validation.go

```diff
@@ -19,16 +19,22 @@ import (
 	apivalidation "k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
-
-	"sigs.k8s.io/kueue/pkg/constants"
 )
 
 var (
 	annotationsPath       = field.NewPath("metadata", "annotations")
-	ParentWorkloadKeyPath = annotationsPath.Key(constants.ParentWorkloadAnnotation)
-	QueueNamePath         = annotationsPath.Key(constants.QueueAnnotation)
+	labelsPath            = field.NewPath("metadata", "labels")
+	parentWorkloadKeyPath = annotationsPath.Key(ParentWorkloadAnnotation)
+	queueNameLabelPath    = labelsPath.Key(QueueLabel)
 )
 
+func ValidateCreateForQueueName(job GenericJob) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, ValidateLabelAsCRDName(job, QueueLabel)...)
+	allErrs = append(allErrs, ValidateAnnotationAsCRDName(job, QueueAnnotation)...)
+	return allErrs
+}
+
 func ValidateAnnotationAsCRDName(job GenericJob, crdNameAnnotation string) field.ErrorList {
```

Comment:
can be unexported now?

Reply:
Still is used by parent-workload, could follow up as for Queuename